pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -123,10 +128,26 @@ type cloudWatchExecutor struct {
 	cfg         *setting.Cfg
 	sessions    SessionCache
 	regionCache sync.Map
+	logger      log.Logger
 
 	resourceHandler backend.CallResourceHandler
 }
 
+// instrumentContext adds plugin key-values to the context; later, logger.FromContext(ctx) will provide a logger
+// that adds these values to its output.
+// TODO: this really shouldn't be here!
+func instrumentContext(ctx context.Context, endpoint string, pCtx backend.PluginContext) context.Context {
+	p := []any{"endpoint", endpoint, "pluginId", pCtx.PluginID}
+	if pCtx.DataSourceInstanceSettings != nil {
+		p = append(p, "dsName", pCtx.DataSourceInstanceSettings.Name)
+		p = append(p, "dsUID", pCtx.DataSourceInstanceSettings.UID)
+	}
+	if pCtx.User != nil {
+		p = append(p, "uname", pCtx.User.Login)
+	}
+	return log.WithContextualAttributes(ctx, p)
+}
+
```

Comment:
This duplicates the function of the same name from `services/pluginsintegration/clientmiddleware`. That function is still being applied, but the sdk's log functions can't reach the context value it's adding, which is (as is generally suggested) private. Obviously this doesn't really belong here, but fixing that is a bigger problem and this works for now.